sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -573,17 +590,31 @@ def fit(self, X, y, sample_weight=None):
                         X_binned_small_train,
                         y_small_train,
                         sample_weight_small_train,
+                        indices_small_train,
                     ) = self._get_small_trainset(
-                        X_binned_train, y_train, sample_weight_train, self._random_seed
+                        X_binned_train,
+                        y_train,
+                        sample_weight_train,
+                        self._random_seed,
                     )
 
+                    # scoring is a predefined metric string
+                    if isinstance(self.scoring, str):
+                        raw_predictions_small_train = raw_predictions[
+                            indices_small_train
+                        ]
```

Comment:
Shouln't we use the whole training set as in `self.scoring == "loss"`?

Reply:
Yea, this is still the case. This branch is never reached when `self.scoring == "loss"`. 

That being said, I can see how using `instance(..., str)` can lead to confusion. I updated this PR with [`89c3dc0` (#26163)](https://github.com/scikit-learn/scikit-learn/pull/26163/commits/89c3dc072a5ddb293e3472f77ff98ed8ecb2d776) to have a more explicit variable name for "scorer is a predefined string".